### DISCUSSION TITLE
Possible Future Documentation Topics

### DISCUSSION THREAD

**Discussion by jhillairet** [scikit-rf/scikit-rf#645] [Ideas]
I'm listing here a list of topics that could fit into the documentation. This list is a never-ending list, things I have in mind, and will be updated regularly and enriched with any proposals:

 - [ ] Measuring/Determining an impedance using a VNA. 
 Ressources : [Three Methods to Measure Impedance with the NanoVNA](https://www.youtube.com/watch?v=1UbEz73FGCU&t=333s),  [Agilent note p.4](https://studylib.net/doc/18887795/impedance-measurements)
- [ ] Introduction to Mixed-Mode S-Parameters.
Ressources: [Mixed-Mode S-Parameters](https://www.microwaves101.com/encyclopedias/mixed-mode-S-parameters), 
- [ ] Make it more clear between characteristic and reference impedances of Network and Media. 
Ressources: [Matching Circuits Reference Impedance?](https://www.researchgate.net/post/Matching-Circuits-Reference-Impedance)
- [ ] Amateur Radio & SDR oriented stuffs. 
Ressources: [W0QE Videos](https://www.youtube.com/channel/UCKSyLSu4fm_1RHoO3Jvk4YQ),  [https://www.rtl-sdr.com/](https://www.rtl-sdr.com/), [Baluns](https://www.microwaves101.com/encyclopedias/baluns) or [Optimum Lossy Broadband Matching Networks for Resonant Antennas](https://www.arrl.org/files/file/Antenna%20Book%20Supplemental%20Files/23rd%20Edition/Optimum%20Lossy%20Broadband%20Matching%20Networks%20for%20Resonant%20Antennas%20-%20AI1H%20-%20QEX%20Sep%202008.pdf)
- [ ] Various ways to create RF components
Ressources: [Top 7 Ways to Create a Quadrature (90˚) Phase Shift](https://www.markimicrowave.com/blog/top-7-ways-to-create-a-quadrature-90-phase-shift/), [6 Ways to Make an N-Way Power Splitter](https://www.markimicrowave.com/blog/how-to-make-an-n-way-power-splitter/)
- [ ] more de-embedding class documentation examples ?
- [ ] Determine Two-Port S-Parameters from One-Port Measurements. 
Ressources: paper from Ou, J., Caggiano, M.F., 2005. https://doi.org/10/czwqwj

**Comment by jhillairet** [comment]
@vik-s it would be great if you could make a notebook example and write some tests about the de-embbeding class #480

**Comment by vik-s** [comment]
> 
> 
> @vik-s it would be great if you could make a notebook example and write some tests about the de-embbeding class #480

Thanks for the followup @jhillairet! I have the notebook example written up more than half-way, I need to finish it up. It's going up on my to-do list.

**Comment by biergaizi** [comment]
My proposal: Solving the complex output impedance of a signal generator with only scalar power measurements and different load impedances. This is a real problem I've encountered in metrology, I think it would be quite interesting to demonstrate how it works in theory using scikit-rf and scipy. I'll write the document for the "Metrology" section and open a pull request when I have time.

Below is a summary of how the method works:

In RF calibration, knowing the output impedance (or reflection coefficient, or VSWR) of a signal generator is extremely important. However, the output of many RF generators in metrology is actively regulated via Automatic Level/Gain Control and active feedback - such as a leveled sinewave generator used for RF power and oscilloscopes calibration, or the 0 dBm, 50 MHz reference output on an RF power meter - Since the level/gain of the generator is sampled by a power splitter or coupler then regulated by the feedback, theoretically its effective output impedance or reflection coefficient is only a property of the coupling device, and it's NOT a property of the amplifier itself. "Effective" VSWR can be as low as a perfect 1.00. But it’s a nonlinear circuit and cannot be measured with standard methods. The Federal Office of Metrology in Switzerland [has proposed a new method](http://resource.npl.co.uk/docs/networks/anamet/members_only/meetings/32/20091016_anamet32_furrer.pdf) using a directional coupler, multiple delay lines and solving nonlinear least-square equations on a computer.

Consider the simple case of a signal generator connected to ideal, but deliberately mismatched power meters with known impedances. You do multiple power measurements, each with a different mismatch. If you write down the equations that describe the power delivered into the power meter, then each RF power measurement defines a circle on the complex plane for all the possible reflection coefficients, and the complex impedance is the intersection point of these circles, which can be found by numerically solving a system of nonlinear equations (well, in this particular case, each RF power measurement doesn't really define a single circle but set of infinitely many possible circles, but you get the point...)

Practically, a "deliberately mismatched power meter" is usually infeasible, since the power delivered to the meter approaches 0 at some frequencies, the error is too large to be practical.  METAS's method is a further refinement of the method. Instead of using a mismatched power meter, it uses a directional coupler with mismatched loads attached, with a power meter to sample the forward power to the load. To simplify the math a bit, each time a mismatched load is attached, the coupler is fully characterized by a VNA as a 2-port network, not a 3-port.

This method is essentially an ad-hoc version of 6-port VNA.

## References

* [Traceable Source Match Calibration of RF & MW Generators](http://resource.npl.co.uk/docs/networks/anamet/members_only/meetings/32/20091016_anamet32_furrer.pdf), 32st ANAMET Meeting, Jürg Furrer, Swiss Federal Office of Metrology (METAS)
* [Power Splitter Characterisation – EM Day](http://resource.npl.co.uk/docs/networks/electromagnetics/071129/rfmt_howes/miall.pdf), James Miall, National Physical Laboratory, UK

**Reply by jhillairet** [reply]
Great topic!
